mule-extensions-api/src/main/java/org/mule/runtime/extension/internal/loader/enricher/OAuthDeclarationEnricher.java

```diff
@@ -108,7 +108,7 @@ public void enrich(ExtensionLoadingContext extensionLoadingContext) {
     final Set<Integer> visitedOwners = new HashSet<>();
     final Set<Integer> visitedProviders = new HashSet<>();
     final Reference<ConnectionProviderDeclaration> ocsConnectionProvider = new Reference<>(null);
-    final boolean ocsEnabled = isPlatformManagedOAuthEnabled();
+    final boolean ocsEnabled = isPlatformManagedOAuthEnabled() || extensionLoadingContext.isOCSEnabled();
```

Comment:
is `isPlatformManagedOAuthEnabled()` reading a system property? If so i think that should be done at the earliest point possible when creating the extensionLoadingContext

Reply:
I set it at `DefaultExtensionLoadingContext` now, but should we rely on that being the only way of creating an `ExtensionLoadingContext` although it's annotated with `@NoImplement`?